src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -281,6 +297,76 @@ public bool IsSolutionOpen
             }
         }
 
+        public async Task<Tuple<bool, bool>> CheckSolutionUIPreConditionsAsync()
```

Comment:
instead of Tuple, please use two out params as it makes for more readable code. i can't figure out from the method definition what the purpose of this tuple is.